src/Microsoft.VisualStudio.Threading/ReentrantSemaphore.cs

```diff
@@ -143,7 +142,12 @@ public static ReentrantSemaphore Create(int initialCount = 1, JoinableTaskContex
         /// <summary>
         /// Executes a given operation within the semaphore.
         /// </summary>
-        /// <param name="operation">The delegate to invoke once the semaphore is entered.</param>
+        /// <param name="operation">
+        /// The delegate to invoke once the semaphore is entered. Not guaranteed to be executed on the context this was originally called on.
+        /// If a <see cref="JoinableTaskContext"/> was supplied to the constructor, this delegate will execute on the main thread if  is
+        /// invoked on the main thread, otherwise it will be invoked on the threadpool. When no <see cref="JoinableTaskContext"/> is supplied
+        /// to the constructor, this delegate will execute on the caller's context or the task scheduler.
```

Comment:
>context or the task scheduler [](start = 75, length = 29)

the caller's task scheduler is part of their context. So I'd leave it at just "the caller's context" or "the caller's SynchronizationContext or TaskScheduler".